In [2]:
import pandas as pd

In [3]:
# Read data in
df = pd.read_csv('input.csv')
df.head()

Set #         Name  Normal/Holo  Rev  Rev - M
0    #1    Bulbasaur            6  NaN      1.0
1    #2      Ivysaur            4  1.0      NaN
2    #3  Venusaur ex            2  NaN      NaN
3    #4   Charmander            7  1.0      NaN
4    #5   Charmeleon            3  NaN      1.0

In [4]:
# fill nulls with 0
df.fillna(value=0, inplace=True)

In [5]:
# change column type to ints
df['Rev'] = df['Rev'].astype('int64')
df['Rev - M'] = df['Rev - M'].astype('int64')

In [6]:
# 1st condition, remove ex
# 2nd condition, 
filtered_df = df[~df['Name'].str.contains('ex')].loc[0:164]
filtered_df.head()

Set #        Name  Normal/Holo  Rev  Rev - M
0    #1   Bulbasaur            6    0        1
1    #2     Ivysaur            4    1        0
3    #4  Charmander            7    1        0
4    #5  Charmeleon            3    0        1
6    #7    Squirtle            6    1        0

In [7]:
# As of 7/30/2024, these are the cards I still need to complete the 151 sv2a set
missing_rev_df = filtered_df[filtered_df['Rev'] == 0]
missing_rev_df

Set #        Name  Normal/Holo  Rev  Rev - M
0      #1   Bulbasaur            6    0        1
4      #5  Charmeleon            3    0        1
18    #19     Rattata            5    0        1
46    #47    Parasect            2    0        1
117  #118     Goldeen            9    0        1
148  #149   Dragonite            4    0        0

In [8]:
mast_df = filtered_df[filtered_df['Rev - M'] != 0]
mast_df.head()

Set #        Name  Normal/Holo  Rev  Rev - M
0     #1   Bulbasaur            6    0        1
4     #5  Charmeleon            3    0        1
7     #8   Wartortle            6    2        1
12   #13      Weedle            7    2        1
18   #19     Rattata            5    0        1

# Cards Needed:

### Reverse (Pokeball) Holos:
* \#1 Bulbasaur
* \#5 Charmeleon
* \#19 Rattata
* \#47 Parasect
* \#118 Goldeen
* \#149 Dragonite

### Masterball Holos:
Too many to list